### ISSUE TITLE
BigQuery: Issues With Parsing Quotation Marks Around String Literal Aliases Used in UNPIVOT Operations

### ISSUE THREAD

**Issue by kdw2126** [sqlfluff/sqlfluff#2616]
## Expected Behaviour
Whenever valid BigQuery SQL code is run through ```sqlfluff lint```, it does not produce unexpected parse errors.   Since literals can either be quoted using single- or double-quotation marks according to BigQuery documentation, this suggests that changing string literal aliases from single to double quotation marks should not trigger parsing errors if the underlying SQL is still valid.

In addition, in contexts where variable names have to be given as string literals in BigQuery, SQLFluff should understand that these items need to be quoted and should not produce the L059 error.

## Observed Behaviour
When row_value_alias values used in an UNPIVOT operation in BigQuery (see definition [here](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#unpivot_operator)) are quoted using double-quotation marks rather than single-quotation marks a PARSE error is produced.

When row_value_alias values used in an UNPIVOT operation in BigQuery (see definition [here](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#unpivot_operator)) are quoted using single-quotation marks, SQLFluff suggests removing quotation marks around the alias even though they are required in this context.

## Steps to Reproduce
(1). Save the following (nonsense but valid) jinja/dbt SQL file:
```sql
    SELECT
        *
    FROM model
    UNPIVOT(
      (A, B)
      FOR year
      IN ((C, D) AS "year_2011", (E, F) AS "year_2012"))
```
(2). Set BigQuery as the dialect and jinja as the templater and run sqlfluff lint on this file.  You will get a parse error.

(3).  Change the file as follows and re-run and the parse error disappears, confirming the issue is with the double-quotation marks used as string literal names.
```sql
    SELECT
        *
    FROM model
    UNPIVOT(
      (A, B)
      FOR year
      IN ((C, D) AS 'year_2011', (E, F) AS 'year_2012'))
```

However, this now produces L059 errors:
- Unnecessary quoted identifier 'year_2011'.
- Unnecessary quoted identifier 'year_2012'.

These errors are also erroneous because these alias [must be string literals](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#rules_for_row_value_alias), meaning that they have to be quoted in this context.  Very confusing for BigQuery to use different rules for aliases for UNPIVOT operations but that's the case here unfortunately.

## Dialect
BigQuery

## Version
SQLFluff version is 0.10.0.  Python version is 3.8.10.
I'm using dbt 1.0.1 but the same issue occurs when no templater is used.

## Configuration
```
[sqlfluff]
dialect = bigquery
exclude_rules = L003,L008,L011,L014,L016,L029,L031,L034

[sqlfluff:rules]
max_line_length = 120
comma_style = leading

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper
```

**Comment by kdw2126** [comment]
Hah, after writing the ticket I found another similar bug so I updated my description to document both.  Please let me know if issues like this should be split up instead!
